packages/test/src/integration-tests.ts

```diff
@@ -1121,4 +1122,64 @@ export function runIntegrationTests(codec?: PayloadCodec): void {
     });
     t.pass();
   });
+
+  if ('promiseHooks' in v8) {
+    // Skip in old node versions
+    test('Stack trace query returns stack that makes sense', async (t) => {
+      const { client } = t.context;
+      const workflowId = uuid4();
+      const rawStacks = await client.execute(workflows.stackTracer, {
+        taskQueue: 'test',
+        workflowId,
+      });
+      const [stack1, stack2] = rawStacks.map((r) =>
+        r
+          .split('\n\n')
+          .map((s) => cleanOptionalStackTrace(`\n${s}`))
```

Comment:
Since the workflow is specifically for this test, could be worth keeping lines, but not necessary of course

Reply:
I considered this but some of the lines originate in framework code and it'll be annoying to have to updates this test anytime it changes.